pkg/provisioner/crdprovisioner.go

```diff
@@ -314,9 +314,14 @@ func (c *CrdProvisioner) DeleteVolume(ctx context.Context, volumeID string, secr
 		return err
 	}
 
-	// if volume deletion already in process, return to prevent duplicate request
-	if azVolumeInstance.Status.State == v1alpha1.VolumeDeleting {
+	switch azVolumeInstance.Status.State {
+	case v1alpha1.VolumeDeleting:
+		// if volume deletion already in process, return to prevent duplicate request
 		return status.Errorf(codes.Aborted, "deletion still in process for volume (%s)", volumeName)
+	case v1alpha1.VolumeDeletionFailed:
+		// if volume deletion state is DeletionFailed, do not retry
+		// TO DO: figure out why the volume deletion is failing and fix it.
+		return nil
```

Comment:
I believe we *do* want to requeue the deletion attempt. There appears to be a missing state transition in the [allowedTargetVolumeStates](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/be7a2db911cf6d8f7a8ce984c6ac41294a712a61/pkg/controller/azvolume.go#L74) map from `DeletionFailed` to `Deleting`.

Reply:
The disk was successfully detached (see the discussion in #206), so I suspect that CRP signalled completion before updating its internal state and the delete request came in before it was able to do so.

Reply:
> I believe we _do_ want to requeue the deletion attempt. There appears to be a missing state transition in the [allowedTargetVolumeStates](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/be7a2db911cf6d8f7a8ce984c6ac41294a712a61/pkg/controller/azvolume.go#L74) map from `DeletionFailed` to `Deleting`.

Ah, I thought you were suggesting that we block retry on failed volume deletion as a temporary getaway for the problem we were having.

Reply:
Sorry for the confusion. I had assumed the phantom attachment may have been due to a temporary condition, and the failure to have a state transition from `DeletionFailed` was causing us to prevent deletion while reporting the same (stale) error over and over.